src/Nethermind/Nethermind.Serialization.Rlp/ConsensusRequestDecoder.cs

```diff
@@ -0,0 +1,120 @@
+// SPDX-FileCopyrightText: 2024 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System;
+using Nethermind.Core.ConsensusRequests;
+
+namespace Nethermind.Serialization.Rlp;
+
+public class ConsensusRequestDecoder : IRlpStreamDecoder<ConsensusRequest>, IRlpValueDecoder<ConsensusRequest>, IRlpObjectDecoder<ConsensusRequest>
+{
+    private readonly WithdrawalRequestDecoder _withdrawalRequestDecoder = new();
+    private readonly DepositDecoder _depositDecoder = new();
+    public int GetContentLength(ConsensusRequest item, RlpBehaviors rlpBehaviors)
+    {
+        int length = item.Type switch
+        {
+            ConsensusRequestsType.WithdrawalRequest => _withdrawalRequestDecoder.GetContentLength((WithdrawalRequest)item, rlpBehaviors),
+            ConsensusRequestsType.Deposit => _depositDecoder.GetContentLength((Deposit)item, rlpBehaviors),
+            _ => throw new RlpException($"Unsupported consensus request type {item.Type}")
+        };
+        return length;
+    }
+
+    public int GetLength(ConsensusRequest item, RlpBehaviors rlpBehaviors)
+    {
+        int length = item.Type switch
+        {
```

Comment:
in GetLength we could call GetContentLength of ConsensusRequest to avoid extra switch here (check TxDecoder